packages/jest-cli/src/TestRunner.js

```diff
@@ -376,12 +380,74 @@ class TestRunner {
       this.addReporter(new CoverageReporter());
     }
 
-    this.addReporter(new SummaryReporter());
     if (config.notify) {
       this.addReporter(new NotifyReporter(this._startRun));
     }
   }
 
+  /**
+   * _setupDefaultReporters
+   * Method for setting up the default reporters
+   */
+  _setupDefaultReporters() {
+    const config = this._config;
+
+    this.addReporter(
+      config.verbose
+        ? new VerboseReporter(config)
+        : new DefaultReporter()
+    );
+
+    this.addReporter(new SummaryReporter());
+  }
+
+  /**
+   * Add Custom reporters to Jest
+   * Custom reporters can be added to Jest using the reporters option in Jest
+   * Config. The format for adding a custom reporter is following
+   *
+   * "reporters": [
+   *    ["reporterPath/packageName", { option1: 'fasklj' }], 
+   *    // Format if we want to specify options
+   *    
+   *    "reporterName"
+   *    // Format if we don't want to specify any options
+   * ]
+   */
+  _addCustomReporters(reporters: any) {
+    let reporterPath, reporterConfig;
+
+    reporters.forEach(entry => {
+      if (Array.isArray(entry)) {
+        [reporterPath, reporterConfig] = entry;
+      } else {
+        if (typeof entry !== 'string') {
+          throw new Error(`
+            Unexpected custom reporter configuration.
+            Expected to get either a path string or an array of [path, confg]
+            got: ${JSON.stringify(entry)}
+          `);
+        }
+
+        reporterPath = entry;
+      }
+
+      try {
+        const reporter = require(reporterPath);
+        this.addReporter(new reporter(reporterConfig || {}));
+      } catch (error) {
+        console.error(`
+          Failed to set up reporter:
+            ${JSON.stringify(reporterPath)}
+          Config:
+            ${JSON.stringify(reporterConfig)}
+        `);
```

Comment:
Can we print this similarly to how we are already printing warnings/errors, with colors etc? I don't think you need JSON.stringify for the reporterPath, for example.